tests/Services/SearchServiceTest.php

```diff
@@ -155,6 +156,177 @@ public function testRedirectToProductPageOnDoSearch(
         $searchServiceMock->doSearch($requestMock, $searchQueryMock);
     }
 
+    public function productProvider(): array
+    {
+        return [
+            'One product found' => [
+                'responseVariationIds' => [
+                    1011, 1012
+                ],
+                'itemSearchServiceResultsAll' => [
+                    'success' => true,
+                    'total' => 1,
+                    'documents' => [
+                        [
+                            'id' => 1011
+                        ]
+                    ]
+                ],
+                'attributes' => []
+            ],
+            'Multiple products found' => [
+                'responseVariationIds' => [
+                    1011, 1022, 1023
+                ],
+                'itemSearchServiceResultsAll' => [
+                    'success' => true,
+                    'total' => 2,
+                    'documents' => [
+                        [
+                            'id' => 1011
+                        ],
+                        [
+                            'id' => 1022
+                        ]
+                    ]
+                ],
+                'attributes' => []
+            ],
+            'One product found and query string type is corrected' => [
+                'responseVariationIds' => [
+                    1011, 1022, 1023
+                ],
+                'itemSearchServiceResultsAll' => [
+                    'success' => true,
+                    'total' => 1,
+                    'documents' => [
+                        [
+                            'id' => 1011
+                        ]
+                    ]
+                ],
+                'attributes' => []
+            ],
+            'One product found and query string type is improved' => [
```

Comment:
Isn't that the same scenario as ` 'One product found and query string type is corrected'`?

Reply:
It seems that query string type differentiation is not making any difference, since `'One product found'` only differs from that number of variation ids

Reply:
Yeah they were actually the same. I actually copied that provider and removed the stuff that I didn't need. This caused that duplicated scenarios. They're removed now ^^.